chia/server/introducer_peers.py

```diff
@@ -1,31 +1,52 @@
 import random
 import time
-from typing import Dict, List, Optional
+from typing import Set, List, Optional
+from dataclasses import dataclass
 
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.peer_info import PeerInfo
-from chia.util.ints import uint64
+from chia.util.ints import uint64, uint16
 
 
+@dataclass(frozen=False)
+class VettedPeer:
+    host: str
+    port: uint16
+
+    # True means we have vetted and it was alive
+    # False means we have not vetted this peer (recent enough)
+    vetted: bool = False
+    vetted_timestamp: uint64 = 0
+    time_added: uint64 = 0
+
+    def __init__(self, h: str, p: uint16):
+        self.host = h
+        self.port = p
+
+    def __eq__(self, rhs):
+        return self.host == rhs.host and self.port == rhs.port
+
+    def __hash__(self):
+        return hash((self.host, self.port))
+
 class IntroducerPeers:
     """
     Has the list of known full node peers that are already connected or may be
     connected to, and the time that they were last added.
     """
 
     def __init__(self) -> None:
-        self._peers: List[PeerInfo] = []
-        self.time_added: Dict[bytes32, uint64] = {}
+        self._peers: Set[VettedPeer] = set()
```

Comment:
this is another change, use a Set instead of a list for the peers. This should make the `in` operator run faster for large peer lists.